rose_lm_node:
    ros__parameters:
        lidar_topic: /velodyne_points                  # 点云话题名
        imu_topic: /livox/imu                      # IMU话题名
        lidar_type: velodyne             # 雷达类型 livox_custom_msg, livox_pointcloud2, custom_mid360_driver, unilidar
        lidar_frame: front_mid360                  # 雷达坐标系
        base_frame: base_footprint
        robot_base_frame: gimbal_yaw
        save_pcd: false                           # 是否保存点云
        small_point_lio:
            # 点云过滤
            min_distance: 0.5                          # 点云半径最小值，小于该值会被过滤
            max_distance: 1000.0                       # 点云半径最大值，大于该值会被过滤
            point_filter_num: 1
            space_downsample_leaf_size: 0.2            # 点云降采样时使用的栅格大小
            batch_interval: 0.0001
            # IMU处理
            gravity: [0.0, 0.0, -9.810]                # 重力
            fix_gravity_direction: true                # 是否收集前200个IMU数据修正重力方向，重力的大小依旧从gravity参数获得
            check_satu: true                           # 是否开启IMU的数据饱和检查
            satu_acc: 30.0                              # IMU的加速度饱和阈值
            satu_gyro: 35.0                            # IMU的角速度饱和阈值
            acc_norm: 9.81                             # IMU的加速度模长

            # 地图
            map_resolution: 0.2                        # 地图分辨率
            init_map_size: 100                          # 接收到多少个点才初始化地图

            # 雷达与IMU相对位姿
            extrinsic_est_en: false                   # 雷达与IMU的相对位姿是否通过预测得到
            extrinsic_T: [ 0.0, 0.0, 0.0 ]
            extrinsic_R: [ 1.0, 0.0, 0.0,
                                0.0, 1.0, 0.0,
                                0.0, 0.0, 1.0 ]

            # 滤波器参数
            max_iter: 3
            # R
            laser_point_cov: 0.001                      # 激光点的协方差
            imu_meas_acc_cov: 0.001                     # IMU测量的加速度协方差
            imu_meas_omg_cov: 0.001                     # IMU测量的角速度协方差
            # Q
            velocity_cov: 20.0                         # 速度的协方差
            acceleration_cov: 500.0                    # 加速度的协方差
            omg_cov: 1000.0                            # 角速度的协方差
            ba_cov: 0.0001                             # 加速度偏置的协方差
            bg_cov: 0.0001                             # 角速度偏置的协方差
            plane_threshold: 0.1                       # 平面匹配阈值，越小表示越严格
            match_sqaured: 81.0                        # 当前点是否在平面上的阈值，越小表示越严格
    